### ISSUE TITLE
Editing the Occtype field in the Missing Data window (opened from the Structure Inventory importer) caused exception.

### ISSUE THREAD

**Issue by jbslaughter** [HydrologicEngineeringCenter/HEC-FDA#882]
Build dated 03/01/2023
Selecting an Occupancy Type attribute with missing fields (just to test) opens the Missing Data window. Two issues in the Missing Data box - the Foundation Height, Ground Elevation, and Structure Value columns did not populate. Also, editing the "Missing" data column for a cell (in the Occupancy Types column) and closing the window caused an exception.

![image](https://github.com/HydrologicEngineeringCenter/HEC-FDA/assets/89167848/d2f35bea-3e0e-4b84-8c8e-cf3c35bc47fc)

Exception Message
Cannot set Owner property to a Window that has been closed.

Call Stack
   at System.Windows.Window.set_Owner(Window value)
   at HEC.FDA.View.ViewWindow.WindowSpawner(WindowVM newvm, Boolean asDialogue) in C:\Code\C#\FDA\HEC-FDA\View\ViewWindow.xaml.cs:line 50
   at HEC.FDA.ViewModel.Utilities.WindowVM.CurrentView_RequestNavigation(IDynamicTab tab, Boolean newWindow, Boolean asDialog) in C:\Code\C#\FDA\HEC-FDA\ViewModel\Utilities\WindowVM.cs:line 87
   at HEC.FDA.ViewModel.BaseViewModel.Navigate(IDynamicTab tab, Boolean newWindow, Boolean asDialog) in C:\Code\C#\FDA\HEC-FDA\ViewModel\BaseViewModel.cs:line 246
   at HEC.FDA.ViewModel.Inventory.ImportStructuresFromShapefileVM.CheckForMissingValues() in C:\Code\C#\FDA\HEC-FDA\ViewModel\Inventory\ImportStructuresFromShapefileVM.cs:line 144
   at HEC.FDA.ViewModel.Inventory.ImportStructuresFromShapefileVM.NextButtonClicked() in C:\Code\C#\FDA\HEC-FDA\ViewModel\Inventory\ImportStructuresFromShapefileVM.cs:line 171
   at HEC.FDA.View.Inventory.ImportStructuresFromShapefile.NextButton_Click(Object sender, RoutedEventArgs e) in C:\Code\C#\FDA\HEC-FDA\View\Inventory\ImportStructuresFromShapefile.xaml.cs:line 23
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.Controls.Primitives.ButtonBase.OnClick()
   at System.Windows.Controls.Button.OnClick()
   at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

**Comment by rnugent3** [comment]
Thanks for reporting @jbslaughter 

@codymccoy Please take a look

**Comment by codymccoy** [comment]
I was unable to reproduce the exception. I agree that the Missing Data UI is confusing. I should make the table not editable. If i remember correctly the UI exists only to show the user what values are missing. It is not meant to fill those missing values in. I think that would have to be done using another program like Qgis or arcgis. The other confusing part is that the plank cells mean that those values are not missing, the cells with "Missing" have missing values. Sort of weird because if it is blank, then it has a value, if it has a value, then it is blank.
![image](https://github.com/HydrologicEngineeringCenter/HEC-FDA/assets/47757848/f6842449-e06d-486b-b3cc-d273402712af)

Not sure how hard it would be but maybe we should fill out the values for each row and leave the missing values blank.

**Comment by rnugent3** [comment]
Finding the values and placing them in the table would be ideal. 

Displaying "not missing" would work if necessary

**Comment by codymccoy** [comment]
@rnugent3 This ended up being a bit harder than I thought it would be (what isn't), but I have all the row data displaying. 
![image](https://github.com/HydrologicEngineeringCenter/HEC-FDA/assets/47757848/0bc7f828-e460-4488-b659-7348b9c450b4)

In our labels at the top, should we mention that the user will need to fill these value in by editing the structures in gis software and then try importing again?

I tried to add a cell styling that would highlight the missing fields but couldn't figure out how to do that with the way I did this datagrid. I also added to the column headers the actual selection that the user made within parenthesis. I figured that would be helpful because that is the column the user will need to look in when in gis software like Qgis. 
Let me know if you want anything changed.
